commons-hpcc/src/main/java/org/hpccsystems/commons/ecl/FileFilter.java

```diff
@@ -16,44 +16,111 @@
 package org.hpccsystems.commons.ecl;
 
 import java.io.Serializable;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.hpccsystems.commons.filter.SQLExpression;
+import org.hpccsystems.commons.filter.SQLFilter;
+import org.hpccsystems.commons.filter.SQLOperator;
+import org.hpccsystems.commons.filter.SQLFragment.FragmentType;
+import org.hpccsystems.commons.filter.SQLOperator.OperatorType;
+import org.json.JSONArray;
+import org.json.JSONObject;
 
 /**
  * A filter to select records from a file or key. The filter is conjunction of
  * field filters with each field having a list of one or more value ranges.
  *
+ * filter : setfilter | wildfilter ;
+    setfilter :  field equals set
+    wildfilter : field wildcard
+    set         : setfragment (, setfragment)*
+    setfragment : ( upperinclusive | upperexclusive) setvalue ( lowerinclusive | lowerexclusive) //ignore the single paren notation for now
+    setvalue    : INTEGER_NUM | QUOTEDUTF8STR
+    QUOTEDUTF8STR : squote utf8string squote;
+    upperinclusive    : '[' ;
+    lowerinclusive    : ']' ;
+    upperexclusive    : '(' ;
+    lowerexclusive    : ')' ;
+    wildcard : '*' ;
+    equals   : '=' ;
+    range    : ':' ;
+    squote   : '\'' ;
+    field : ( 'A'..'Z' | 'a'..'z') (( 'A'..'Z' | 'a'..'z') | UNDERSCORE | ETC | INTEGER_NUM  )* INTEGER_NUM)?;
+    INTEGER_NUM : ( '1'..'9')( '0'..'9')*;
+ *
  */
 public class FileFilter implements Serializable
```

Comment:
Since we cannot do grouping at the moment, this doesn't matter but once we have grouping support I think it would make sense to have FileFilter be a tree structure. This would be needed to represent more complex filters like: "Age > 18 && ( CarColor == "Blue" || CarMake == "Tesla")

Reply:
Agreed... https://track.hpccsystems.com/browse/JAPI-205
Once you report the grouping issue to the platform, let's add that jira to japi-205